python/test/test_property_graph.py

```diff
@@ -314,3 +314,60 @@ def func_operator(g, prop, out, nid):
     assert oprop[0].as_py() == 0
     assert oprop[4].as_py() == 0
     assert oprop[-1].as_py() == 0
+
+
+def test_types(graph):
+    node_type_set = set()
+    edge_type_set = set()
+    for nid in range(0, graph.num_nodes()):
+        node_type_set.add(graph.get_type_of_node(nid))
+        for eid in graph.edge_ids(nid):
+            edge_type_set.add(graph.get_type_of_edge(eid))
+
+    assert len(node_type_set) == 11
+    assert len(edge_type_set) == 15
+
+    assert graph.does_node_have_type(0, 17)
+    assert graph.does_edge_have_type(0, 8)
+
+    node_atomic_types = graph.node_types.atomic_types
+    node_name_to_id_map = {name: node_atomic_types[name].type_id for name in node_atomic_types}
+    assert node_name_to_id_map == {
+        b"Tag": 12,
```

Comment:
Using the numeric IDs in the test will be fragile since we don't really guarantee stability of the numeric ID (I assume). This doesn't need to be fixed in this PR, but it is something to keep in mind.

Reply:
Confirming this: I've been told many times by others that the stability of a GID is not guaranteed.
I've been doing my ID tests using numeric ID properties instead because of that (it's
a lot more painful to do it that way but it's better than having all these tests break if the
way GIDs work suddenly changes elsewhere).

Reply:
@bowu please address this somehow. In this case I think checking everything by name would work fine, so that seems like the way to go.